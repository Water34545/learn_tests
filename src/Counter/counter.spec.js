import React from "react";
import Counter from "./counter";

const setUp = () => shallow(<Counter />);

describe('Counter component', () => {
  let component
  let instance

  beforeEach(() => {
    component = setUp();
    instance = component.instance()
  })

  it('should render Counter component without props', () => {
    expect(component).toMatchSnapshot()
  })

  describe('Counter handlers', () => {
    it('should change count value to plus 1', () => {
      const btn = component.find('.plusOneBtn')
      btn.simulate('click')
      expect(component.state().count).toBe(1)
    })

    it('should reset counter value to 10', () => {
      const btn = component.find('.resetBtn')
      btn.simulate('click')
      expect(component.state().count).toBe(10)
    })

    it('should reset counter value to 20', () => {
      instance.handleReset(20)
      expect(component.state().count).toBe(20)
    })
  })
})