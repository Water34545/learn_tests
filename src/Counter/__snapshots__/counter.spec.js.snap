// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should render Counter component without props 1`] = `
<div>
  <div>
    0
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;
